src/hatch/env/virtual.py

```diff
@@ -364,8 +363,10 @@ def _patched_propose_interpreters(*args, **kwargs):
 
         virtualenv_discovery.propose_interpreters = _patched_propose_interpreters
         try:
+            installed_python = self.python_manager.get_installed().get(python_version)
+            try_first_with = (installed_python.python_path,) if installed_python else ()
             python_info = virtualenv_discovery.get_interpreter(
-                python_version, (), env=self.get_interpreter_resolver_env()
+                python_version, try_first_with, env=self.get_interpreter_resolver_env()
```

Comment:
Can you please tell me the reason once more for this change? I'm quite hesitant because it breaks (as far as I can tell) the contract that I specify in the docs: https://hatch.pypa.io/latest/plugins/environment/virtual/#python-resolution

Basically, we want to check (by default) every external Python followed by the internally managed ones.

I think you mentioned something about the current logic hiding exceptions, can you tell me about that so that it may be fixed instead?

Reply:
Sure - I was wondering why we did python resolution by frobbing `$PATH` in this way, but when you explain it that way it makes perfect sense.  I will revert this change, I have I think a better solution, described below.

Yes, the problem is that `virtualenv.discovery.builtin` does not report exceptions for Python binaries that it finds in `$PATH` and there is no way to change this behaviour: https://github.com/pypa/virtualenv/blob/main/src/virtualenv/discovery/builtin.py#L155

I will report this as a bug against `virtualenv`, for `hatch` what we can do is to improve the error message here to mention that an incompatible version might be lurking in `~/.local/share/hatch/env/virtual/.pythons`: https://github.com/pypa/hatch/blob/master/src/hatch/env/virtual.py#L259

Actually there may be a way to detect the problem - what I noticed is that `_find_existing_interpreter` will return `None`, because virtualenv discovery has silently failed, but `PythonManager.get_installed` will still return the broken python version.  So instead of breaking the contract to get virtualenv discovery to do the right thing we can just check for any installed distributions in `check_compatibility` in the case where `_find_existing_interpreter` returns  `None`, and warn about the situation.

Reply:
Also, I think, any such fix should be a separate PR, so I changed the title of this one!

Reply:
Also ... I won't actually report this as a bug in `virtualenv` because honestly the behaviour makes sense.  A bad python in `$PATH` *should* just be ignored since there might always be another one somewhere else.